internal/destination/destination.go

```diff
@@ -39,84 +41,18 @@ func NewDestination() sdk.Destination {
 	return sdk.DestinationWithMiddleware(&Destination{}, sdk.DefaultDestinationMiddleware()...)
 }
 
-func (d *Destination) Parameters() map[string]sdk.Parameter {
-	return map[string]sdk.Parameter{
-		config.KeyURLs: {
-			Default:     "",
-			Required:    true,
-			Description: "The connection URLs pointed to NATS instances.",
-		},
-		config.KeySubject: {
-			Default:     "",
-			Required:    true,
-			Description: "A name of a subject to which the connector should write.",
-		},
-		config.KeyConnectionName: {
-			Default:     "",
-			Required:    false,
-			Description: "Optional connection name which will come in handy when it comes to monitoring.",
-		},
-		config.KeyNKeyPath: {
-			Default:     "",
-			Required:    false,
-			Description: "A path pointed to a NKey pair.",
-		},
-		config.KeyCredentialsFilePath: {
-			Default:     "",
-			Required:    false,
-			Description: "A path pointed to a credentials file.",
-		},
-		config.KeyTLSClientCertPath: {
-			Default:  "",
-			Required: false,
-			//nolint:lll // long description
-			Description: "A path pointed to a TLS client certificate, must be present if tls.clientPrivateKeyPath field is also present.",
-		},
-		config.KeyTLSClientPrivateKeyPath: {
-			Default:  "",
-			Required: false,
-			//nolint:lll // long description
-			Description: "A path pointed to a TLS client private key, must be present if tls.clientCertPath field is also present.",
-		},
-		config.KeyTLSRootCACertPath: {
-			Default:     "",
-			Required:    false,
-			Description: "A path pointed to a TLS root certificate, provide if you want to verify server’s identity.",
-		},
-		config.KeyMaxReconnects: {
-			Default:  "5",
-			Required: false,
-			Description: "Sets the number of reconnect attempts " +
-				"that will be tried before giving up. If negative, " +
-				"then it will never give up trying to reconnect.",
-		},
-		config.KeyReconnectWait: {
-			Default:  "5s",
-			Required: false,
-			Description: "Sets the time to backoff after attempting a reconnect " +
-				"to a server that we were already connected to previously.",
-		},
-		ConfigKeyRetryWait: {
-			Default:     "5s",
-			Required:    false,
-			Description: "Sets the timeout to wait for a message to be resent, if send fails.",
-		},
-		ConfigKeyRetryAttempts: {
-			Default:     "3",
-			Required:    false,
-			Description: "Sets a numbers of attempts to send a message, if send fails.",
-		},
-	}
+func (d *Destination) Parameters() config.Parameters {
+	return Config{}.Parameters()
```

Comment:
@raulb let me know this is not correct, because of the SDK parameters. We should be using `NewDestination().Parameters()`

Reply:
@raulb After revisiting this, I believe this is correct. We need the `NewDestination().Parameters()` when parsing the configs, which we already use.

Reply:
Yes, using `NewDestination().Paramters()` here would create endless recursion.